[cairo] io2d build failure on Ubuntu

**Host Environment**
- OS: Ubuntu 19.10
- Compiler: GCC 9.2.1

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install io2d

**Failure logs** 
I'm having the same issue when installing io2d on Linux Ubuntu
```
[1/8] /usr/bin/c++  -DCAIRO_HAS_XLIB_SURFACE=1 -I/home/user/vcpkg/buildtrees/io2d/src/3b5b5fc338-1775a03870/P0267_RefImpl/P0267_RefImpl/cairo/xlib -I/home/user/vcpkg/buildtrees/io2d/src/3b5b5fc338-1775a03870/P0267_RefImpl/P0267_RefImpl/cairo -I/home/user/vcpkg/buildtrees/io2d/src/3b5b5fc338-1775a03870/P0267_RefImpl/P0267_RefImpl -isystem /home/user/vcpkg/installed/x64-linux/include -fPIC -g   -std=gnu++17 -MD -MT P0267_RefImpl/P0267_RefImpl/cairo/xlib/CMakeFiles/io2d_cairo_xlib.dir/cairo_renderer_xlib.cpp.o -MF P0267_RefImpl/P0267_RefImpl/cairo/xlib/CMakeFiles/io2d_cairo_xlib.dir/cairo_renderer_xlib.cpp.o.d -o P0267_RefImpl/P0267_RefImpl/cairo/xlib/CMakeFiles/io2d_cairo_xlib.dir/cairo_renderer_xlib.cpp.o -c /home/user/vcpkg/buildtrees/io2d/src/3b5b5fc338-1775a03870/P0267_RefImpl/P0267_RefImpl/cairo/xlib/cairo_renderer_xlib.cpp
FAILED: P0267_RefImpl/P0267_RefImpl/cairo/xlib/CMakeFiles/io2d_cairo_xlib.dir/cairo_renderer_xlib.cpp.o 
/usr/bin/c++  -DCAIRO_HAS_XLIB_SURFACE=1 -I/home/user/vcpkg/buildtrees/io2d/src/3b5b5fc338-1775a03870/P0267_RefImpl/P0267_RefImpl/cairo/xlib -I/home/user/vcpkg/buildtrees/io2d/src/3b5b5fc338-1775a03870/P0267_RefImpl/P0267_RefImpl/cairo -I/home/user/vcpkg/buildtrees/io2d/src/3b5b5fc338-1775a03870/P0267_RefImpl/P0267_RefImpl -isystem /home/user/vcpkg/installed/x64-linux/include -fPIC -g   -std=gnu++17 -MD -MT P0267_RefImpl/P0267_RefImpl/cairo/xlib/CMakeFiles/io2d_cairo_xlib.dir/cairo_renderer_xlib.cpp.o -MF P0267_RefImpl/P0267_RefImpl/cairo/xlib/CMakeFiles/io2d_cairo_xlib.dir/cairo_renderer_xlib.cpp.o.d -o P0267_RefImpl/P0267_RefImpl/cairo/xlib/CMakeFiles/io2d_cairo_xlib.dir/cairo_renderer_xlib.cpp.o -c /home/user/vcpkg/buildtrees/io2d/src/3b5b5fc338-1775a03870/P0267_RefImpl/P0267_RefImpl/cairo/xlib/cairo_renderer_xlib.cpp
In file included from /home/user/vcpkg/buildtrees/io2d/src/3b5b5fc338-1775a03870/P0267_RefImpl/P0267_RefImpl/cairo/xlib/xio2d_cairo_xlib_main.h:9,
                 from /home/user/vcpkg/buildtrees/io2d/src/3b5b5fc338-1775a03870/P0267_RefImpl/P0267_RefImpl/cairo/xlib/cairo_renderer_xlib.cpp:1:
/home/user/vcpkg/buildtrees/io2d/src/3b5b5fc338-1775a03870/P0267_RefImpl/P0267_RefImpl/cairo/xlib/xio2d_cairo_xlib_output_surfaces.h:7:10: fatal error: cairo-xlib.h: No such file or directory
 #include <cairo-xlib.h>
          ^~~~~~~~~~~~~~
compilation terminated.
```


https://github.com/microsoft/vcpkg/issues/6619

--- a/ports/cairo/CMakeLists.txt
+++ b/ports/cairo/CMakeLists.txt
@@ -7,6 +7,24 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
   include_directories("./win32")
 endif()
 
+set(CAIRO_HEADERS
+    cairo.h
+    cairo-deprecated.h
+    cairo-features.h
+    cairo-pdf.h
+    cairo-ps.h
+    cairo-script.h
+    cairo-svg.h
+    ../cairo-version.h
+    cairo-win32.h
+    ../util/cairo-gobject/cairo-gobject.h
+    cairo-ft.h
+)
+
+set(CAIRO_X11_HEADERS
+    cairo-xlib.h
+)
+
 file(GLOB SOURCES
 "cairo-analysis-surface.c"
 "cairo-arc.c"
@@ -154,7 +172,7 @@ file(GLOB PLATFORM_SOURCES_WIN32
 "win32/cairo-win32-font.c"
 )
 
-if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
+if(WIN32)
   list(APPEND SOURCES ${PLATFORM_SOURCES_WIN32})
 endif()
 
@@ -237,6 +255,14 @@ endif()
 
 if (CAIRO_HAS_XLIB_SURFACE)
     file(INSTALL cairo-xlib.h DESTINATION include)
+endif()
+
+install(FILES ${CAIRO_HEADERS} DESTINATION include)
+install(FILES ${CAIRO_HEADERS} DESTINATION include/cairo)
+
+if (WITH_X11)
+    install(FILES ${CAIRO_X11_HEADERS} DESTINATION include)
+    install(FILES ${CAIRO_X11_HEADERS} DESTINATION include/cairo)
 endif()
 
 install(TARGETS cairo cairo-gobject

--- a/ports/cairo/CMakeLists.txt
+++ b/ports/cairo/CMakeLists.txt
@@ -3,7 +3,7 @@ project(cairo C)
 
 # Add include directories
 include_directories(".")
-if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
+if(WIN32)
   include_directories("./win32")
 endif()
 
@@ -195,7 +195,7 @@ add_definitions(
     -DHAVE_FT_GET_X11_FONT_FORMAT=1)
 
 # additional features for macOS
-if((CMAKE_SYSTEM_NAME STREQUAL "Darwin") OR (CMAKE_SYSTEM_NAME STREQUAL "Linux"))
+if(UNIX OR APPLE)
   add_definitions(
       -DHAVE_INTTYPES_H=1
       -DHAVE_STDINT_H=1
@@ -214,7 +214,7 @@ endif()
 
 target_link_libraries(cairo PRIVATE ZLIB::ZLIB PNG::PNG Freetype::Freetype unofficial::pixman::pixman-1 unofficial::fontconfig::fontconfig)
 
-if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
+if(WIN32)
     target_link_libraries(cairo PRIVATE gdi32 msimg32 user32)
 endif()
 

--- a/ports/cairo/CONTROL
+++ b/ports/cairo/CONTROL
@@ -1,5 +1,5 @@
 Source: cairo
-Version: 1.16.0-3
+Version: 1.16.0-4
 Homepage: https://cairographics.org
 Description: Cairo is a 2D graphics library with support for multiple output devices. Currently supported output targets include the X Window System (via both Xlib and XCB), Quartz, Win32, image buffers, PostScript, PDF, and SVG file output. Experimental backends include OpenGL, BeOS, OS/2, and DirectFB.
 Build-Depends: zlib, libpng, pixman, glib, freetype, fontconfig

--- a/ports/cairo/portfile.cmake
+++ b/ports/cairo/portfile.cmake
@@ -1,4 +1,3 @@
-include(vcpkg_common_functions)
 set(CAIRO_VERSION 1.16.0)
 
 vcpkg_download_distfile(ARCHIVE
@@ -39,22 +38,7 @@ vcpkg_install_cmake()
 
 vcpkg_fixup_cmake_targets(CONFIG_PATH share/unofficial-cairo TARGET_PATH share/unofficial-cairo)
 
-# Copy the appropriate header files.
-foreach(FILE
-"${SOURCE_PATH}/src/cairo.h"
-"${SOURCE_PATH}/src/cairo-deprecated.h"
-"${SOURCE_PATH}/src/cairo-features.h"
-"${SOURCE_PATH}/src/cairo-pdf.h"
-"${SOURCE_PATH}/src/cairo-ps.h"
-"${SOURCE_PATH}/src/cairo-script.h"
-"${SOURCE_PATH}/src/cairo-svg.h"
-"${SOURCE_PATH}/cairo-version.h"
-"${SOURCE_PATH}/src/cairo-win32.h"
-"${SOURCE_PATH}/util/cairo-gobject/cairo-gobject.h"
-"${SOURCE_PATH}/src/cairo-ft.h")
-  file(COPY ${FILE} DESTINATION ${CURRENT_PACKAGES_DIR}/include)
-  file(COPY ${FILE} DESTINATION ${CURRENT_PACKAGES_DIR}/include/cairo)
-endforeach()
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 
 foreach(FILE "${CURRENT_PACKAGES_DIR}/include/cairo.h" "${CURRENT_PACKAGES_DIR}/include/cairo/cairo.h")
     file(READ ${FILE} CAIRO_H)
@@ -67,9 +51,8 @@ foreach(FILE "${CURRENT_PACKAGES_DIR}/include/cairo.h" "${CURRENT_PACKAGES_DIR}/
 endforeach()
 
 # Handle copyright
-file(COPY ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/cairo)
-file(RENAME ${CURRENT_PACKAGES_DIR}/share/cairo/COPYING ${CURRENT_PACKAGES_DIR}/share/cairo/copyright)
+file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
 
 vcpkg_copy_pdbs()
 
-vcpkg_test_cmake(PACKAGE_NAME unofficial-cairo)
+#vcpkg_test_cmake(PACKAGE_NAME unofficial-cairo)

--- a/scripts/ci.baseline.txt
+++ b/scripts/ci.baseline.txt
@@ -640,7 +640,6 @@ intel-mkl:x86-windows=fail
 intelrdfpmathlib:arm-uwp=fail
 intelrdfpmathlib:x64-linux=fail
 intelrdfpmathlib:x64-uwp=fail
-io2d:x64-linux=fail
 irrlicht:arm64-windows=fail
 irrlicht:arm-uwp=fail
 irrlicht:x64-osx=fail

